src/Clapi/TextSerialisation.hs

```diff
@@ -0,0 +1,33 @@
+module Clapi.TextSerialisation where
+import Data.Monoid
+import qualified Data.Map as Map
+import Blaze.ByteString.Builder (Builder)
+import Blaze.ByteString.Builder.Char.Utf8 (fromString, fromChar)
+import qualified Clapi.Serialisation as Wire
+import Clapi.Types (Time(..), ClapiValue(..))
+
+typeTag :: ClapiValue -> Char
+typeTag (ClBool _) = 'B'
+typeTag v = Wire.typeTag v
+
+fss :: Show a => a -> Builder
+fss a = fromString $ show a
+
+cvBuilder :: ClapiValue -> Builder
+cvBuilder (ClBool True) = fromChar 'T'
+cvBuilder (ClBool False) = fromChar 'F'
+cvBuilder (ClInt32 i) = fss i
+cvBuilder (ClString s) = fss s
+
+tpBuilder :: (Time, [ClapiValue]) -> Builder
+tpBuilder ((Time s f), vs) = tb <> mconcat (map (\v -> (fromChar ' ') <> (cvBuilder v)) vs)
+  where
+    tb = (fss s) <> (fromChar ':') <> (fss f)
+
+encode :: Map.Map Time [ClapiValue] -> Builder
+encode ts = header <> bodyBuilder
+  where
+    -- It is invalid for a time series to be empty, so this use of head is
+    -- kinda fine, but the errors will suck:
+    header = fromString $ map typeTag $ head $ Map.elems ts
```

Comment:
`map :: (a -> b) -> [a] -> [b]`
`fmap :: (Functor t) => (a -> b) -> t a -> t b`
`fmap` is more general, and I tend to use it everywhere out of habit. I think in Idris they "fixed" this, and it's just called `map` again, but in Haskell functors were late to the party!

Because of laziness, you could also do `head $ Map.elems $ fmap typeTag ts`. That might sometimes be a nicer way to think about the data structures. None of this is worth it, just thinking out loud.